utils/azure_users.py

```diff
@@ -119,6 +146,21 @@ def to_azure_user(self, item) -> AzureUser:
         role = item[self.role_field] if there_is_role else None
         return AzureUser(id, name, email, role)
 
+    def to_azure_user_v2(self, item) -> AzureUser:
+        there_is_email = len(item['otherMails']) > 0
+
+        id = item['objectId']
+        name = item['displayName']
+        email = item['otherMails'][0] if there_is_email else ''
+        roles = []
+        for role_id in ROLE_FIELD_VALUES.keys():
+            field_name, _ = ROLE_FIELD_VALUES[role_id]
+            if field_name in item:
+                field_value = item[field_name]
+                roles.append(field_value)
```

Comment:
Can we use something more sophisticated like `filter` or `reduce`, maybe `lamba functions` or `comprehension lists`

Reply:
Sure :)

Reply:
Thanks for the suggestion. I found two ways to do it 
```
field_names = [field_name for (field_name, _)  in ROLE_FIELD_VALUES.values()]
field_names_in_item = filter(lambda field_name: field_name in item , field_names)
roles = map(lambda field_name: item[field_name], field_names_in_item)
roles = list(roles)

or

roles = [
    item[field_name] for (field_name, field_value) in ROLE_FIELD_VALUES.values()
    if field_name in item
]

```

I think the second one in cleaner.